src/gossip_agent/waku_handling.rs

```diff
@@ -66,37 +68,44 @@ pub fn filter_peer_subscriptions(
     node_handle: &WakuNodeHandle<Running>,
     graphcast_topic: &Option<WakuPubSubTopic>,
     content_topics: &[WakuContentTopic],
-) -> Result<Vec<String>, Box<dyn Error>> {
+) -> Result<Vec<String>, WakuHandlingError> {
     let subscription: FilterSubscription =
         content_filter_subscription(graphcast_topic, content_topics);
-    let filter_subscribe_result: Vec<String> = node_handle
-        .peers()?
-        .iter()
-        .filter(|&peer| {
-            peer.peer_id().as_str()
-                != node_handle
-                    .peer_id()
-                    .expect("Failed to find local node's peer id")
-                    .as_str()
-        })
-        .map(|peer: &WakuPeerData| {
-            let filter_res = node_handle.filter_subscribe(
-                &subscription,
-                peer.peer_id().clone(),
-                Duration::new(6000, 0),
-            );
-            match filter_res {
-                Ok(_) => format!(
-                    "Filter subcription request made to: {} - {:#?}",
-                    peer.peer_id(),
-                    &subscription
-                ),
-                Err(e) => format!("Failed to filter subscribe with peer: {e}"),
-            }
-        })
-        .collect();
-    info!("Filter subscription added: {:#?}", filter_subscribe_result);
-    Ok(filter_subscribe_result)
+
+    let peers = node_handle.peers();
```

Comment:
Similar to an earlier comment of avoid multi-level nested statements just for matching errors from a different function. I think an example like this should work fine and easier to read later on 
```
let filter_subscribe_result: Vec<String> = node_handle
    .peers().map_err(|e| WakuHandlingError::UnableToRetrievePeers(e))?    // <-----------------
    .iter()
    .filter(|&peer| {
        peer.peer_id().as_str()
            != node_handle
                .peer_id()
                .expect("Failed to find local node's peer id")
                .as_str()
    })
    .map(|peer: &WakuPeerData| {
        let filter_res = node_handle.filter_subscribe(
            &subscription,
            peer.peer_id().clone(),
            Duration::new(6000, 0),
        );
        match filter_res {
            Ok(_) => format!(
                "Filter subcription request made to: {} - {:#?}",
                peer.peer_id(),
                &subscription
            ),
            Err(e) => format!("Failed to filter subscribe with peer: {e}"),
        }
    })
    .collect();
    info!("Filter subscription added: {:#?}", filter_subscribe_result);
    Ok(filter_subscribe_result)
```